File: src/Functions.Worker.Extensions.OpenAI/Embeddings/InputType.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

namespace Microsoft.Azure.Functions.Worker.Extensions.OpenAI.Embeddings;

/// <summary>
/// Options for interpreting input binding data.
/// </summary>
public enum InputType
{
    /// <summary>
    /// The input data is raw text.
    /// </summary>
    RawText,

    /// <summary>
    /// The input data is a file path that contains the text.
    /// </summary>
    FilePath,

    /// <summary>
    /// The input data is a URL that contains the text.
    /// </summary>
    URL
}

File: java-library/src/main/java/com/microsoft/azure/functions/openai/annotation/embeddings/InputType.java
/**
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 */

package com.microsoft.azure.functions.openai.annotation.embeddings;


/**
 * <p>
 * Options for interpreting embeddings input binding data.
 * </p>
 */
public enum InputType {
    /**
     * The input data is raw text.
     */
    RawText,

    /**
     * The input data is a file path that contains the text.
     */
    FilePath,

    /**
     * The input data is a URL.
     */
    Url
}
